packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -22,20 +22,28 @@ import { SyntheticShadowRoot } from '../../faux-shadow/shadow-root';
 
 export function pathComposer(startNode: EventTarget, composed: boolean): EventTarget[] {
     const composedPath: EventTarget[] = [];
-    let current: EventTarget | null = startNode;
-    const startRoot: Window | Node =
-        startNode instanceof Window ? startNode : (startNode as Node).getRootNode();
+
+    let startRoot: Window | Node;
+    if (startNode instanceof Window) {
+        startRoot = startNode;
+    } else if (startNode instanceof Node) {
+        startRoot = startNode.getRootNode();
+    } else {
+        return composedPath;
+    }
+
+    let current: Window | Node | null = startNode;
     while (!isNull(current)) {
         composedPath.push(current);
 
-        if (current instanceof Element) {
-            // Text also implements the Slottable mixin but is not an EventTarget, nor a ParentNode,
-            // which means it is safe to only look for Elements here.
-            const assignedSlot: HTMLSlotElement | null = current.assignedSlot;
+        if ((current as Element | Text).assignedSlot) {
```

Comment:
Any reason why not doing `current instanceof Element || current instanceof Text` ?